### DISCUSSION TITLE
Adobe Animate .aab build

### DISCUSSION THREAD

**Discussion by ghost** [airsdk/Adobe-Runtime-Support#927] [Q&A] [answered]
Okay, so. With the new .533 release...

Is it possible to create an .aab file exporting an .apk file with animate and uncomment the code line inside adt.cfg?~ (you know, the code line who creates an.aab file).

Should It work that way? 

And... If we use an ANE inside, should that ANE work properly too?

**Comment by ajwfrost** [comment, marked as answer]
Hi

Yes exactly: if you're using Animate, or another IDE where you can't select an "AAB" output type yet, then build as if you're doing an APK but uncomment the line so that you have `CreateAndroidAppBundle=true` in your config file.

So then it's basically as if the request from Animate was switched from `-target apk -arch whatever` to `-target aab` with the rest of the params unchanged.

ANEs should work fine now because we've changed how we're doing these, the classes are being compiled and linked properly with Gradle rather than just the separate resource-bundling and code-linking that we had done before (where it lost the ability to reference resources in ANEs via the 'R.strings.id' sort of mechanism).

The only thing to watch out for might be that an AAB includes all four CPU types: armv7, armv8, x86 and x86_64. So someone with a x86_64 device can now get the proper native support when they download your app: but if the ANE only included armv7 and armv8 support then they'll not get the platform-specific ANE functionality, they'll just get the 'default' code.

thanks

**Reply by P2b10** [reply]
Hi Andrew:

1. Just to be sure, if I am using Adobe Animate AIR SDK 33.1.1.575 all I need to do is uncomment this line #CreateAndroidAppBundle=true?

2. If I am using Adobe Animate 2015 in order to be able to compile with a 1024 certificate should I do the same, but using the armv8 arch?

Best

**Comment by idanasher** [comment]
Regarding this issue.

I am also using animate to compile, for years now.

Do I have to use this new aat feature ? 
Or is it only apply to large apps that uses bundle ?

**I am about to distribute my new app apk to the store, 
compiled with animate and air the beta  33.1.1.533 SDK and using latest distriqt ane's.** 

Do I have to be worry about this crashing on some devices going to production with this now ? 

Thanks Andrew & Everybody

**Comment by leossmith** [comment]
@idanasher in August 2021 this will become mandatory for all new apps and in November 2021 for updating older apps.

**Comment by Gokulv617** [comment]
I'm using the latest version of Adobe Animate v21.0.7 but there is no option to select .aab publishing format in the UI. Please let me know when this option will be available(like they did for creating 64 bit apk in UI itself) in ANIMATE ?

![img_20210628](https://user-images.githubusercontent.com/43432782/123599395-2c0d9d00-d813-11eb-9f4a-b18ef258149d.png)

**Comment by idanasher** [comment]
@Gokulv617 See andrew's post https://github.com/airsdk/Adobe-Runtime-Support/issues/925#issuecomment-869107708

**Comment by ajwfrost** [comment]
And for completeness: I don't know when Adobe Animate might be updated to add in the UI for this, they're aware of the change here but of course will need time to react to this and implement an update within their standard release cycle. Hence the workaround mentioned above...

**Comment by Gokulv617** [comment]
> @Gokulv617 See andrew's post [#925 (comment)](https://github.com/airsdk/Adobe-Runtime-Support/issues/925#issuecomment-869107708)

yes will try that also. thanks

**Comment by Gokulv617** [comment]
> And for completeness: I don't know when Adobe Animate might be updated to add in the UI for this, they're aware of the change here but of course will need time to react to this and implement an update within their standard release cycle. Hence the workaround mentioned above...

okay. Thanks for updating me

**Comment by idanasher** [comment]
I have uploaded my new Android version about a week ago,
and I'm getting some new crashes\ANRs.

compiled with animate and air the beta 33.1.1.533 SDK and using latest distriqt ane's.
( Did NOT used the new .aab feature yet )

The main and most significant crash stack trace from google play console:

```
Type java.lang.RuntimeException

java.lang.RuntimeException: 
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3967)
  at android.app.ActivityThread.access$1500 (ActivityThread.java:233)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2006)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:241)
  at android.app.ActivityThread.main (ActivityThread.java:7582)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:941)
Caused by: java.lang.ClassNotFoundException: 
  at dalvik.system.BaseDexClassLoader.findClass (BaseDexClassLoader.java:196)
  at java.lang.ClassLoader.loadClass (ClassLoader.java:379)
  at java.lang.ClassLoader.loadClass (ClassLoader.java:312)
  at android.app.AppComponentFactory.instantiateReceiver (AppComponentFactory.java:110)
  at androidx.core.app.CoreComponentFactory.instantiateReceiver (CoreComponentFactory.java:60)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3960)

```

It happens on Android 10/11 devices, did not manage to reproduced it.

**Reply by ajwfrost** [reply]
Does it not even say what class it was that wasn't found?!
`Caused by: java.lang.ClassNotFoundException: `
normally would have the missing class name added to it...

Are you able to tell, does this happen right at start-up?
Are you getting logcat reports i.e. if we added in some notifications to logcat on what's loading when, would you then be able to see these in the error reports that you're receiving?

thanks

**Comment by idanasher** [comment]
Hi Andrew, How are you ?

The logs here are form the google play console.
I dont know when it happens.

I do have some more info:

The classNotFound is refering to :
```
java.lang.ClassNotFoundException
dalvik.system.BaseDexClassLoader.findClass

```

**Reply by ajwfrost** [reply]
Just checking - you've edited this, does that mean it's happening on both 64-bit and 32-bit APKs? and it's happening when you generate APK files rather than Bundles? I'm wondering whether there is an issue with androidx support, but it's also possibly an issue with how the multidex stuff now works.

Is this just "some" crashes, or is it consistent? Would you be able to send your APK file to us for inspecting, so that we can see what classes are provided in which dex files?

thanks

**Comment by idanasher** [comment]
I think it happens o both.

It happen when I generate apk files (not Bundles)

It is pretty consistence since the last version when it has started.

sure, I can send you the apk - how ?
do you have a link for the Harman transporting files system ?

**Reply by ajwfrost** [reply]
Please use: https://transfer.harman.com/requests/yzQK8qYklSZNjuP1hmhqx8
thanks

**Comment by aalarcon0075-commits** [comment]
Could someone send me an .APK download for Adobe animate? The type is Android 11.

**Reply by ajwfrost** [reply]
Not sure what you're after here..? Adobe Animate isn't an Android application, it's a desktop (at least Windows + macOS) application, but it can generate APK files for Android if it uses the AIR SDK. What are you trying to do?

**Reply by aalarcon0075-commits** [reply]
I'm trying to get the apk that someone generated so i can make a pbs kids animation on android and put it on google drive.

**Reply by aalarcon0075-commits** [reply]
I'm trying to Find a generated .APK, I mean like unofficial, so I can import fla. files.

**Reply by ajwfrost** [reply]
I'm not completely sure that first one makes sense to me. But if you want to import fla files, yes you need Animate, but it needs to be on Windows/macOS, you can download it from Adobe's website: https://www.adobe.com/products/animate/free-trial-download.html

**Reply by aalarcon0075-commits** [reply]
It's because I don't know how to explain it. What I was saying is if someone has a unofficial .APK of Adobe animate I can download.

**Reply by ajwfrost** [reply]
I don't think Adobe Animate is available as an APK file (or AAB).. I'm 99% sure it doesn't run on Android...

**Reply by aalarcon0075-commits** [reply]
UUUUGHH!!!! I GIVE UP! I WAS THIS CLOSE TO BE MAKING ANIMATONS AND YOU TELLED ME THERE'S NO APK FILE?!?!?!? I DONT HAVE A MACOS!

**Reply by ajwfrost** [reply]
Do you have a Windows computer? If not, then yes, you have no chance of using Animate.
If you want to make animations directly using Android, then again, Animate isn't the right tool afaik.
If you want to use a Windows machine and make animations you can then export so that they will run on Android .. that's where all this comes in.

**Reply by aalarcon0075-commits** [reply]
Is there an APK where I can import fla files and animating than Adobe animate?

**Reply by aalarcon0075-commits** [reply]
(cries)

**Reply by marchbold** [reply]
@aalarcon0075-commits Can't say I've tried it but there's this fla web viewer : https://lifeart.github.io/fla-viewer/

Fla's are generally seen as source for an animation that is exported or used in other applications (AIR / webgl / video), and are usually considered unique to Animate. 

What exactly are you trying to achieve? If you are trying to create animations on an Android device for use in an AIR application, you could use something like [VectorMotion](https://play.google.com/store/apps/details?id=app.vectormotion.mobile) and use the sprite sheets or svg outputs.

**Comment by aalarcon0075-commits** [comment]
Every time I go to https://transfer.harman.com/requests/yzQK8qYklSZNjuP1hmhqx8, it says "File Request Expired". Can you help?

**Reply by ajwfrost** [reply]
That was a link to allow someone to share an APK with us, for us to check what was going wrong with it. So yes, that's expired, but that's not an issue..?
